trieste/models/model_interfaces.py

```diff
@@ -27,7 +27,7 @@ class ProbabilisticModel(ABC):
     """ A probabilistic model. """
 
     @abstractmethod
-    def predict(self, query_points: QueryPoints) -> Tuple[ObserverEvaluations, TensorType]:
+    def predict(self, query_points: QueryPoints, full_cov: bool = False) -> Tuple[ObserverEvaluations, TensorType]:
```

Comment:
from previous PR: still under discussion if we should use a `full_cov` argument or a separated `predict_full_cov`